packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -87,6 +87,7 @@ export abstract class InstrumentationBase<T = any>
   ): T {
     if (!baseDir) {
       if (typeof module.patch === 'function') {
+        module.moduleExports = exports;
```

Comment:
why this, can you describe the case you have to use it ?,

Reply:
The case without `baseDir` is for core module and in this case you didn't assigned the `moduleExports` which were used for the `unpatch` callback